src/tools/cephfs/top/cephfs-top

```diff
@@ -499,83 +596,177 @@ class FSTop(object):
             self.refresh_window_size()
-            if self.width <= 2 or self.width <= 2:
+            if self.height <= 4 or self.width <= 2:
                 self.exit_ev.wait(timeout=self.refresh_interval_secs)
                 continue
 
             # coordinate constants for windowing -- (height, width, y, x)
             # NOTE: requires initscr() call before accessing COLS, LINES.
             try:
-                HEADER_WINDOW_COORD = (2, self.width - 1, 0, 0)
+                HEADER_WINDOW_COORD = (4, self.width - 1, 0, 0)
                 self.header = curses.newwin(*HEADER_WINDOW_COORD)
-                if self.height >= 3:
-                    TOPLINE_WINDOW_COORD = (1, self.width - 1, 3, 0)
-                    self.topl = curses.newwin(*TOPLINE_WINDOW_COORD)
+            except curses.error:
+                # this may happen when creating the sub windows the
+                # terminal window size changed, just retry it
+                continue
+
+            global fs_list
+            fs_list = self.get_fs_names()
+            stats_json = self.perf_stats_query()
+            self.flag = 0
+            help_text = " [Press 'q' to go back to home (all filesystem info) screen"\
+                        " and 'm' to select another filesystem]"
+            try:
+                if fs not in fs_list:
+                    self.stdscr.refresh()
+                    headline = "INFO: Selected filesystem is not available now" + help_text
+                    self.refresh_header(stats_json, headline)
                 else:
-                    self.topl = None
-                if self.height >= 5:
-                    MAIN_WINDOW_COORD = (self.height - 4, self.width - 1, 4, 0)
-                    self.mainw = curses.newwin(*MAIN_WINDOW_COORD)
+                    if self.height >= 5:
+                        TOPLINE_WINDOW_COORD = (3, self.width - 1, 4, 0)
+                        self.topl = curses.newwin(*TOPLINE_WINDOW_COORD)
+                    else:
+                        self.topl = None
+                    if self.height >= 8:
+                        MAIN_WINDOW_COORD = (self.height - 7, self.width - 1, 7, 0)
+                        self.mainw = curses.newwin(*MAIN_WINDOW_COORD)
+                    else:
+                        self.mainw = None
+                    headline = "HELP: Selected filesystem info" + help_text
+                    if self.refresh_header(stats_json, headline):
+                        x_coord_map = self.refresh_top_line_and_build_coord()
+                        self.refresh_main_window(x_coord_map, stats_json, fs)
+                curses.halfdelay(self.refresh_interval_secs)
+                c = self.header.getch()
+                if c in [ord('m'), ord('q')]:
+                    self.set_option(c)
                 else:
-                    self.mainw = None
+                    continue
+                self.exit_ev.wait(timeout=self.refresh_interval_secs)
```

Comment:
I've dropped the below code in run_display() and in run_all_display() and There is no change in the execution. Everything the same. So can you explain why this code is necessary?

```
                else:
                    continue
                self.exit_ev.wait(timeout=self.refresh_interval_secs)
```

Reply:
I guess `self.exit_ev.wait(timeout=self.refresh_interval_secs)` is related with the multithreading concept to wait for timeout value to block the thread until timeout occurs, but this will cause the delay in response when multiple wrong keys are pressed continuously...that's why put `else` condition to prevent that case.